packages/react-dom-bindings/src/client/ReactDOMInput.js

```diff
@@ -418,26 +417,37 @@ export function restoreControlledInputState(element: Element, props: Object) {
   }
 }
 
-// In Chrome, assigning defaultValue to certain input types triggers input validation.
-// For number inputs, the display value loses trailing decimal points. For email inputs,
-// Chrome raises "The specified value <x> is not a valid email address".
-//
-// Here we check to see if the defaultValue has actually changed, avoiding these problems
-// when the user is inputting text
-//
-// https://github.com/facebook/react/issues/7253
+// Here we check to see if the defaultValue has actually changed and update it
 export function setDefaultValue(
   node: HTMLInputElement,
   type: ?string,
-  value: ToStringValue,
+  value: mixed,
+  force: ?boolean,
 ) {
   if (
-    // Focused number inputs synchronize on blur. See ChangeEventPlugin.js
-    type !== 'number' ||
-    getActiveElement(node.ownerDocument) !== node
+    type === 'button' ||
+    type === 'file' ||
+    type === 'hidden' ||
+    type === 'image' ||
+    type === 'reset' ||
+    type === 'submit'
   ) {
-    if (node.defaultValue !== toString(value)) {
-      node.defaultValue = toString(value);
-    }
+    return;
+  }
+
+  // In Chrome, assigning defaultValue to certain input types triggers input validation.
+  // For number inputs, the display value loses trailing decimal points. For email inputs,
+  // Chrome raises "The specified value <x> is not a valid email address".
+  // https://github.com/facebook/react/issues/7253
+  if (
+    (type === 'number' || type === 'email') &&
```

Comment:
does `url` have the same issue?

Reply:
To be honest, I don't know. Even the number input doesn't have this problem now, so I based it on the previous comment (_I think in about a year, this difference for number input can be removed_)